cpu/stm32/include/periph_cpu.h

```diff
@@ -1174,6 +1174,57 @@ void stm32_eth_common_init(void);
 #define HAVE_PTP_TIMER_SET_ABSOLUTE 1   /**< Native implementation available */
 /** @} */
 
+/**
+ * @brief   Backup RAM must be initialized with @ref backup_ram_init on reset
+ */
+#define BACKUP_RAM_HAS_INIT         1
+
+/**
+ * @brief   Enable backup RAM access
+ */
+#if IS_USED(MODULE_PERIPH_BACKUP_RAM)
+void backup_ram_init(void);
+#else
+static inline void backup_ram_init(void) {}
+#endif
+
+/**
+ * @brief   Check whether the backup RAM regulator is on
+ */
+#if IS_USED(MODULE_PERIPH_BACKUP_RAM)
+bool backup_ram_regulator_is_on(void);
+#else
+static inline bool backup_ram_regulator_is_on(void) {return false;}
+#endif
+
+/**
+ * @brief   Enable the backup RAM regulator to retain it´s content
+ *          during standby and VBat mode
+ */
+#if IS_USED(MODULE_PERIPH_BACKUP_RAM)
+void backup_ram_regulator_on(void);
+#else
+static inline void backup_ram_regulator_on(void) {}
+#endif
+
+/**
+ * @brief   Disable the backup RAM regulator
+ */
+#if IS_USED(MODULE_PERIPH_BACKUP_RAM)
+void backup_ram_regulator_off(void);
+#else
+static inline void backup_ram_regulator_off(void) {}
+#endif
+
+/**
+ * @brief   Returns true if the CPU woke up from deep sleep (backup/standby)
+ */
+#if IS_USED(MODULE_PERIPH_BACKUP_RAM)
+bool cpu_woke_from_backup(void);
+#else
+static inline bool cpu_woke_from_backup(void) {return false;}
```

Comment:
Couldn't that still be useful when backup ram is not used?

Reply:
I skimmed over the [cmsis-header-stm32](https://github.com/modm-io/cmsis-header-stm32) repository.
I guess a function for all stm32 would be more complicated. Some don´t have `RCC_CSR_PORRSTF`. Some don´t have `RCC_CSR_BORRSTF`. And `stm32h7` doesn´t have reset flags at all.

I would rename it to `backup_ram_is_retained()` because the use case of this function is to check exactly that.

Reply:
or `backup_ram_was_retained()` because it is called _after_ reset